addons/account/models/account_account.py

```diff
@@ -745,9 +745,12 @@ def _toggle_reconcile_to_true(self):
                 amount_residual = (debit-credit),
                 amount_residual_currency = amount_currency
             WHERE full_reconcile_id IS NULL and account_id IN %s
+            RETURNING id
         """
         self.env.cr.execute(query, [tuple(self.ids)])
-        self.env['account.move.line'].invalidate_model(['amount_residual', 'amount_residual_currency', 'reconciled'])
+        lines = self.env['account.move.line'].browse(id for [id] in self.env.cr.fetchall())
```

Comment:
The `modified()` part looks correct, but ideally `cr.execute()` should be preceded by flushing what the SQL query depends on, something like
```py
self.env['account.move.line'].flush_model(['amount_residual', 'amount_residual_currency', 'reconciled', 'full_reconcile_id', 'account_id'])
```

Reply:
It doesn't depend on the residual amounts or on reconcile 🤔
But those need to be invalidated.
Or are you using the (still mind-blowing imo) fact that invalidating flushes?

Reply:
Invalidation methods automatically flush.  That's a technical requirement, which is due to implementation choices.  It actually makes the implementation correct and simpler.

Reply:
In other words, if there is some data to flush and you don't flush _before_, it's going to be flushed _after_ your SQL query, which might push some _former_ update after this SQL update...

Reply:
Yes, I got that.
But what if I want to just discard the cache value, without doing a query (useless) to flush it? Which is what I want in 90% of the cases where I do a `invalidate_*` instead of a `flush_*`